rasa_nlu/model.py

```diff
@@ -78,17 +80,21 @@ def __init__(self, metadata: Dict[Text, Any], model_dir: Optional[Text]):
     def get(self, property_name, default=None):
         return self.metadata.get(property_name, default)
 
+    def component_name(self, index):
+        return self.get('pipeline', [])[index]['name']
+
     @property
-    def component_classes(self):
-        if self.get('pipeline'):
-            return [c.get("class") for c in self.get('pipeline', [])]
-        else:
-            return []
+    def component_names(self):
```

Comment:
why does this use the `class` property but `component_name` uses the `name` property?

Reply:
legacy 💩

Reply:
it should be back component classes, it is used to validate requirements for these classes